### PR TITLE
template setup

### PR DESCRIPTION
**Pull request by h1alexbel**

### CODE REVIEW

**Comment by h1alexbel** [comment]
@rultor merge

**Comment by l3r8yJ** [comment]
@h1alexbel to merge these changes, you need to create a maven project that is fully built

File: checkstyle-suppressions.xml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+and fix all documentation.
+-->
+<!DOCTYPE suppressions PUBLIC "-//Checkstyle//DTD SuppressionFilter Configuration 1.0//EN" "https://checkstyle.org/dtds/suppressions_1_0.dtd">
+<suppressions>
+  <suppress checks="LineLength" files=".*\.java$" lines="0-23"/>
+  <suppress checks="JavadocType" files=".*\.java$" lines="23-9999"/>
+  <suppress checks="JavadocVariable" files=".*\.java$" lines="23-9999"/>
+  <suppress checks="JavadocMethod" files=".*\.java$" lines="23-9999"/>
+  <suppress checks="HiddenField" files=".*\.java$" lines="23-9999"/>
+  <suppress checks="VisibilityModifier" files="KfFlexibleEnvelope.java" lines="44"/>
```
Comment (l3r8yJ) [inline checkstyle-suppressions.xml:37]:
```suggestion
  
```

File: checkstyle-suppressions.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  <suppress checks="VisibilityModifier" files="KfAttrEnvelope.java" lines="36"/>
```
Comment (l3r8yJ) [inline checkstyle-suppressions.xml:38]:
```suggestion
```

File: checkstyle-suppressions.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  <suppress checks="VisibilityModifier" files="KfAttrEnvelope.java" lines="38"/>
```
Comment (l3r8yJ) [inline checkstyle-suppressions.xml:39]:
```suggestion
```

File: checkstyle-suppressions.xml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+<!--
+@todo #4:60min Turn on JavaDoc suppressions.
+We have to turn on JavaDoc suppressions
+and fix all documentation.
+-->
+<!DOCTYPE suppressions PUBLIC "-//Checkstyle//DTD SuppressionFilter Configuration 1.0//EN" "https://checkstyle.org/dtds/suppressions_1_0.dtd">
+<suppressions>
+  <suppress checks="LineLength" files=".*\.java$" lines="0-23"/>
+  <suppress checks="JavadocType" files=".*\.java$" lines="23-9999"/>
+  <suppress checks="JavadocVariable" files=".*\.java$" lines="23-9999"/>
```
Comment (l3r8yJ) [inline checkstyle-suppressions.xml:34]:
```suggestion
```

File: checkstyle-suppressions.xml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+-->
+<!--
+@todo #4:60min Turn on JavaDoc suppressions.
+We have to turn on JavaDoc suppressions
+and fix all documentation.
+-->
+<!DOCTYPE suppressions PUBLIC "-//Checkstyle//DTD SuppressionFilter Configuration 1.0//EN" "https://checkstyle.org/dtds/suppressions_1_0.dtd">
+<suppressions>
+  <suppress checks="LineLength" files=".*\.java$" lines="0-23"/>
+  <suppress checks="JavadocType" files=".*\.java$" lines="23-9999"/>
```
Comment (l3r8yJ) [inline checkstyle-suppressions.xml:33]:
```suggestion
```

File: checkstyle-suppressions.xml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  <suppress checks="JavadocVariable" files=".*\.java$" lines="23-9999"/>
+  <suppress checks="JavadocMethod" files=".*\.java$" lines="23-9999"/>
```
Comment (l3r8yJ) [inline checkstyle-suppressions.xml:35]:
```suggestion
```

**Review by l3r8yJ — CHANGES_REQUESTED** [review]
@h1alexbel take a look, please 

I noted the suppressions that should be removed

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+          </dependency>
+        </dependencies>
+        <executions>
+          <execution>
+            <id>verify-style</id>
+            <phase>process-classes</phase>
+            <goals>
+              <goal>check</goal>
+            </goals>
+            <configuration>
```
Comment (l3r8yJ) [inline pom.xml:266]:
```suggestion
```

also this `configuration` section should be removed

Comment (h1alexbel) [reply pom.xml:266]: @l3r8yJ I can't, license is longer than 80 lines

Comment (l3r8yJ) [reply pom.xml:266]: @h1alexbel yeah, never mind, my bad

**Comment by h1alexbel** [comment]
@rultor merge

**Comment by h1alexbel** [comment]
@rultor merge

**Comment by h1alexbel** [comment]
@rultor merge

**Pull request merged**